cypress/e2e/pokedex.cy.js

```diff
@@ -0,0 +1,124 @@
+describe("Pokedex testing", () => {
+  const localhost = "http://127.0.0.1:8080/";
+  const firstPage = `https://pokeapi.co/api/v2/pokemon/?limit=20&offset=0`;
+  const secondPage = `https://pokeapi.co/api/v2/pokemon/?limit=20&offset=20`;
+  const lastPage = `https://pokeapi.co/api/v2/pokemon/?limit=20&offset=1280`;
+  const pokemonListURL = "https://pokeapi.co/api/v2/pokemon/?limit=100000&offset=0";
+  const blastoiseURL = "https://pokeapi.co/api/v2/pokemon/9";
+  const blastoiseSpeciesURL = "https://pokeapi.co/api/v2/pokemon-species/blastoise";
+
+  beforeEach(() => {
+    cy.visit(localhost);
+    cy.intercept("GET", firstPage, { fixture: "pokedexFirstPage" }).as("getFirstPage");
+    cy.intercept("GET", pokemonListURL, { fixture: "pokemonList" }).as("getAllPokemons");
+  });
+
+  it("Should load the first page cards", () => {
+    const POKEMONS_PER_PAGE = 20;
+
+    cy.get("[data-cy='pokemon-cards-container']").children().should("have.length", POKEMONS_PER_PAGE);
+    cy.get("[data-cy='previous-page-button']").should("have.class", "disabled");
+    cy.get("[data-cy='next-page-button']").should("have.not.class", "disabled");
+
+    cy.get("[data-cy='page-1']").should("be.visible");
+    cy.get("[data-cy='page-2']").should("be.visible");
+    cy.get("[data-cy='page-3']").should("be.visible");
+    cy.get("[data-cy='page-4']").should("be.visible");
+  });
+
+  it("Uses the pokemon search box", () => {
+    cy.intercept("GET", blastoiseURL, { fixture: "blastoise" }).as("getBlastoise");
+    cy.intercept("GET", blastoiseSpeciesURL, { fixture: "blastoiseSpecies" }).as("getBlastoiseSpecies");
+
+    cy.get("[data-cy='pokemon-search-box']").click();
+    cy.get("[data-cy='pokemon-search-box-list']").children().should("have.length", 1292);
+    cy.get("[data-cy='pokemon-search-box-list']").should("be.visible");
+
+    cy.get("[data-cy='pokemon-search-box']").type("Blastoise");
+    cy.get("[data-cy='Blastoise']").should("be.visible");
+
+    cy.get("[data-cy='Blastoise']").click();
+    cy.get("[data-cy='pokemon-card-modal']").should("be.visible");
+
+    cy.wait(1000);
+    cy.get("[data-cy='pokemon-modal-close-button']").click();
+
+    cy.get("[data-cy='pokemon-search-box-list']").should("not.be.visible");
+
+    cy.get("[data-cy='pokemon-search-box']").click();
+    cy.get("[data-cy='pokemon-search-box-list']").should("be.visible");
+    cy.get("[data-cy='pokemon-search-box']").clear();
+    cy.get("[data-cy='pokemon-search-box']").type("Wartortle");
+
+    cy.get("[data-cy='Wartortle']").should("be.visible");
+  });
+
+  it("Clicks on pokemon card", () => {
```

Comment:
Can you put something more descriptive?

Reply:
Added your feedback to my spec file, thank you !